app/uk/gov/hmrc/merchandiseinbaggage/controllers/Navigator.scala

```diff
@@ -32,17 +32,10 @@ import scala.concurrent.{ExecutionContext, Future}
 class Navigator {
   import NavigatorMapping._
 
-  def nextPage(request: NavigationRequests): Call = request match {
-    case RequestByPass(url)                             => toNextPage(url)
-    case RequestByPassWithIndex(url, idx)               => nextPageWithIndex(idx)(url)
-    case GoodsTypeQuantityRequest(declarationType, idx) => goodsTypeQuantity(declarationType, idx)
-  }
-
-  def nextPageWithCallBack(request: NavigationRequestsAsync)(implicit ec: ExecutionContext): Future[Call] = request match {
-    case r: RequestWithCallBack => reviewGoods(r.value, r.declarationJourney, r.overThresholdCheck, r.callBack)
-    case r: RequestWithIndexAndCallBack =>
-      purchaseDetails(r.purchaseDetailsInput, r.index, r.journey, r.goodsEntry, r.callBack)
-
+  def nextPage(request: NavigationRequestsAsync)
+              (implicit ec: ExecutionContext): Future[Call] = request match {
+    case ReviewGoodsRequest(value, journey, overThresholdCheck, upsert)    => reviewGoods(value, journey, overThresholdCheck, upsert)
```

Comment:
Would it be easier to read if all the matchers passed the match value, ie 
`case request:ReviewGoodsRequest => reviewGoods(request)`